ui/src/app/shared/components/webcomponents/confirmation-modal.component.ts

```diff
@@ -29,37 +34,81 @@ import {FormControl} from "@angular/forms";
       </mat-dialog-actions>
     </div>
     <div *ngIf="modalData?.isPermanentDelete && !modalData.disabled">
-      <div class="pb-25 pt-25">
+      <div *ngIf="modalData?.linkedEntityList?.cachedItems" class="pb-25">
         <div class="fz-20 pb-25 d-flex">
-          <span [translate]="'message.common.confirmation.title_with_name'| translate: {Title: modalData?.title}"></span>
+          <span *ngIf="modalData?.linkedEntityList?.cachedItems?.length > 0" class="fz-15" [innerHTML]="'step_group.delete.impacted.message' | translate : {Testcases:modalData?.linkedEntityList?.cachedItems?.length}"></span>
+          <span *ngIf="modalData?.linkedEntityList?.cachedItems?.length == 0" class="fz-15" [innerHTML]="'step_group.delete.zero_impacted.message' | translate"></span>
           <button
             class="close"
             type="button"
             [matTooltip]="'hint.message.common.close' | translate"
             mat-dialog-close>
           </button>
         </div>
+        <div *ngIf="modalData?.linkedEntityList?.cachedItems.length > 0" >
+          <mat-dialog-content class="m-0 p-0 w-100">
+            <div class="list-content overflow-x-hidden  theme-only-items-scroll delete-confirm-popup">
+              <cdk-virtual-scroll-viewport
```

Comment:
it's better to have a wrapper component for virutal scroll and render contents

Reply:
i not sure how to implement this. i just implemented the dialog modal the same way how it was done in cloud.

Reply:
plz create a component and render children content inside the component to make it reusable